internal/v5/api.go

```diff
@@ -1175,24 +1180,40 @@ func (h *ReqHandler) metaPermWithKey(entity *mongodoc.BaseEntity, id *router.Res
 	return nil, errgo.WithCausef(nil, params.ErrNotFound, "unknown permission")
 }
 
-func (h *ReqHandler) visibleACL(acls mongodoc.ACL) (mongodoc.ACL, error) {
-	if h.auth.User == nil {
-		return mongodoc.ACL{
-			Read: []string{"everyone"},
-			Write: []string{},
-		}, nil
+func (h *ReqHandler) visibleACL(req *http.Request, acls mongodoc.ACL) (mongodoc.ACL, error) {
+	respForEveryone := mongodoc.ACL{
+		Read:  []string{"everyone"},
+		Write: []string{},
 	}
-	ok, err := h.auth.User.Allow(acls.Write)
+	auth := h.auth
+	// Authenticate the user if not authenticated already, in order to check
+	// write permissions as well.
+	if auth.User == nil {
+		var err error
+		auth, err = h.Authenticate(req)
+		if err != nil {
+			// The user cannot authenticate, but the entity was readable by everyone anyway.
+			return respForEveryone, nil
```

Comment:
Do we assume this was readable by everyone because we got this far?

Reply:
Yes.